src/facade/dragonfly_connection.cc

```diff
@@ -1038,17 +1055,20 @@ void Connection::DispatchFiber(util::FiberSocketBase* peer) {
     if (dispatch_q_.size() == 1 && cur_epoch == prev_epoch) {
       ThisFiber::Yield();
       DVLOG(2) << "After yielding to producer, dispatch_q_.size()=" << dispatch_q_.size();
+      if (cc_->conn_closing)
+        break;
     }
     prev_epoch = cur_epoch;
+
     builder->SetBatchMode(dispatch_q_.size() > 1);
 
     // Special case: if the dispatch queue accumulated a big number of commands,
     // we can try to squash them
     // It is only enabled if the threshold is reached and the whole dispatch queue
     // consists only of commands (no pubsub or monitor messages)
     bool squashing_enabled = squashing_threshold > 0;
-    bool threshold_reached = dispatch_q_cmds_count_ > squashing_threshold;
-    bool are_all_plain_cmds = dispatch_q_cmds_count_ == dispatch_q_.size();
+    bool threshold_reached = pending_pipeline_cmd_cnt_ > squashing_threshold;
+    bool are_all_plain_cmds = pending_pipeline_cmd_cnt_ == dispatch_q_.size();
```

Comment:
what are monitor messages? are they PipelineMsg?

Reply:
No, they have their own type `MonitorMessage`. Btw `pending_pipeline_cmd_cnt_` was Romans suggestion 😆 I also think it's okay given it has a comment